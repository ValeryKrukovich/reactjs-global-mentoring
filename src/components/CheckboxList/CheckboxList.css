.checkbox-list {
    border: 0;
}

.checkbox-list__legend {
    display: block;
    margin: 0 0 12px;
    font-weight: 600;
    font-size: 16px;
    line-height: 20px;
    letter-spacing: 1px;
    text-transform: uppercase;
    color: var(--red);
    opacity: 0.8;
}

.checkbox-list__list {
    list-style-type: none;
    margin: 0;
    padding: 0;
}
