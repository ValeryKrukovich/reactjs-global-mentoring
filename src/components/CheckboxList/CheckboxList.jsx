import React from "react";
// import './CheckboxList.css';

export function CheckboxList(props) {
    return (
        <fieldset className='checkbox-list'>
            <legend className='checkbox-list__legend'>
                {props.label}
            </legend>
            <ul className='checkbox-list__list'>
                <li>
                    <input type='checkbox' name={props.name} value='Crime' id='Crime' onChange={props.onChange} />
                    <label htmlFor='Crime'>
                        Crime
                    </label>
                </li>
                <li>
                    <input type='checkbox' name={props.name} value='Documentary' id='Documentary' onChange={props.onChange} />
                    <label htmlFor='Documentary'>
                        Documentary
                    </label>
                </li>
                <li>
                    <input type='checkbox' name={props.name} value='Horror' id='Horror' onChange={props.onChange} />
                    <label htmlFor='Horror'>
                        Horror
                    </label>
                </li>
                <li>
                    <input type='checkbox' name={props.name} value='Comedy' id='Comedy' onChange={props.onChange} />
                    <label htmlFor='Comedy'>
                        Comedy
                    </label>
                </li>
            </ul>
        </fieldset>
    )
}
