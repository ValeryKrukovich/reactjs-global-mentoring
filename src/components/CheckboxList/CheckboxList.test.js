import React from "react";
import { render } from '@testing-library/react';

import { CheckboxList } from "./CheckboxList.jsx";

describe('CheckboxList', () => {
    test('render CheckboxList snapshot', () => {
        const { asFragment } = render(<CheckboxList />);
        expect(asFragment(<CheckboxList />)).toMatchSnapshot();
    });
});