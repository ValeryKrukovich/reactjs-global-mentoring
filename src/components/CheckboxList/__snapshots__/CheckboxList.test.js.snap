// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxList render CheckboxList snapshot 1`] = `
<DocumentFragment>
  <fieldset
    class="checkbox-list"
  >
    <legend
      class="checkbox-list__legend"
    />
    <ul
      class="checkbox-list__list"
    >
      <li>
        <input
          id="Crime"
          type="checkbox"
          value="Crime"
        />
        <label
          for="Crime"
        >
          Crime
        </label>
      </li>
      <li>
        <input
          id="Documentary"
          type="checkbox"
          value="Documentary"
        />
        <label
          for="Documentary"
        >
          Documentary
        </label>
      </li>
      <li>
        <input
          id="Horror"
          type="checkbox"
          value="Horror"
        />
        <label
          for="Horror"
        >
          Horror
        </label>
      </li>
      <li>
        <input
          id="Comedy"
          type="checkbox"
          value="Comedy"
        />
        <label
          for="Comedy"
        >
          Comedy
        </label>
      </li>
    </ul>
  </fieldset>
</DocumentFragment>
`;
